routes/post.go

```diff
@@ -1301,21 +1428,50 @@ func (fes *APIServer) GetSinglePost(ww http.ResponseWriter, req *http.Request) {
 	// Slice the comments from the offset up to either the end of the slice or the offset + limit, whichever is smaller.
 	maxIdx := lib.MinUint32(commentEntryResponseLength, requestData.CommentOffset+requestData.CommentLimit)
 	var comments []*PostEntryResponse
-	if commentEntryResponseLength > requestData.CommentOffset {
+	if commentEntryResponseLength > requestData.CommentOffset && commentLevel == 0 {
 		comments = commentEntryResponseList[requestData.CommentOffset:maxIdx]
+	} else {
+		comments = commentEntryResponseList
+	}
```

Comment:
Can you add a small comment about why the different treatment based on comment level please?